peer-store/in-memory.go

```diff
@@ -47,7 +48,10 @@ func (me *InMemory) GetPeers(ih InfoHash) (ret []krpc.NodeAddr) {
 }
 
 func (me *InMemory) AddPeer(ih InfoHash, na krpc.NodeAddr) {
-	key := string(na.IP)
+	bs := make([]byte, 2)
+	binary.BigEndian.PutUint16(bs, uint16(na.Port))
+	key := append(na.IP, bs...)
```

Comment:
Not sure why you're doing this, but appending to a slice owned by the caller is not a good idea. This can cause data races if appends occur in different locations. I'm also not sure how you plan to use a []byte as a key.

Reply:
Sorry. I want to use `in-memory` peerstore to store peers info. I found that `krpc.NodeAddr` `UnmarshalBinary` function parses IP and Port, but AddPeer function only uses IP as key.